packages/composite-checkout/src/lib/payment-methods/credit-card.js

```diff
@@ -35,13 +35,60 @@ export function CreditCardSubmitButton() {
 	);
 }
 
+export function CreditCardSummary( { id } ) {
+	const localize = useLocalize();
+	const [ paymentData ] = usePaymentData();
+	let PaymentLogo = null;
+
+	if ( paymentData.creditCard && id === 'card' ) {
+		//TODO: Update this with all credit card types we support
+		switch ( Number( paymentData.creditCard.cardNumber[ 0 ] ) ) {
```

Comment:
I don't like that I'm duplicating this logic here but I can't seem to store the brand value in the paymentData object at the same time as the credit card number value over here: https://github.com/Automattic/wp-calypso/pull/37249/files#diff-fff7af4b8a04000b2023baa81fa1e884R26

Reply:
I'm not sure I understand, but things will likely be different when we switch the data model in #37296 so I'll take another look then. In the meantime duplication is ok.

Reply:
I brought this up in our call the other day where I was trying to update the paymentData object twice and so it was getting written and then overwritten.

Reply:
Looks like this isn't an issue anymore.